### PR TITLE
Several minor improvements to AccessScopeChecker

### PR DESCRIPTION
**Pull request by jrose-apple**
Builds on #19332 per @slavapestov's request. No functionality change beyond what's in #19332.

### CODE REVIEW

**Comment by jrose-apple** [comment]
@swift-ci Please test

**Comment by jrose-apple** [comment]
@swift-ci Please smoke test compiler performance

**Review by slavapestov — APPROVED** [review]
Nice, thanks!

File: lib/Sema/TypeCheckAccess.cpp [hunk 1]
```diff
@@ -995,91 +999,114 @@ class UsableFromInlineChecker : public AccessControlCheckerBase,
...
 
+  /// If \p PBD declared stored instance properties in a fixed-contents struct,
+  /// return said struct.
+  static const StructDecl *
+  getFixedLayoutStructContext(const PatternBindingDecl *PBD) {
+    auto *parentStruct = dyn_cast<StructDecl>(PBD->getDeclContext());
+    if (!parentStruct)
+      return nullptr;
+    if (!parentStruct->getAttrs().hasAttribute<FixedLayoutAttr>() ||
+        PBD->isStatic() || PBD->hasStorage()) {
```
Comment (jrose-apple) [inline lib/Sema/TypeCheckAccess.cpp]: Grr, missed a `!` here.

**Comment by jrose-apple** [comment]
@swift-ci Please test

**Pull request merged**